src/group/mls_group/apply_commit.rs

```diff
@@ -76,19 +75,23 @@ impl MlsGroup {
                     Some(kpb) => kpb,
                     None => return Err(ApplyCommitError::MissingOwnKeyPackage),
                 };
-                provisional_tree
-                    .replace_private_tree(ciphersuite, own_kpb, &serialized_context)
-                    .unwrap()
+                Some(
+                    provisional_tree
+                        .replace_private_tree(ciphersuite, own_kpb, &serialized_context)
+                        .unwrap(),
+                )
             } else {
-                provisional_tree
-                    .update_path(sender, &path, &serialized_context)
-                    .unwrap()
+                Some(
+                    provisional_tree
+                        .update_path(sender, &path, &serialized_context)
+                        .unwrap(),
+                )
             }
         } else {
             if path_required_by_commit {
                 return Err(ApplyCommitError::RequiredPathNotFound);
             }
-            Secret::from(zero(ciphersuite.hash_length()))
+            None
```

Comment:
`None` is what will be passed to `hkdf_extract` in the end. In there we'll use the default implementation for `Secret`, wich yields a `vec![]`. That's different from `zero(ciphersuite.hash_length())`.

Reply:
Good catch! It's a bit odd, though, because the PSK, for example, defaults to an empty octet string if there is no proper PSK.